### PR TITLE
KZOO-376: disallow deleting the master account db

### PR DESCRIPTION
**Pull request by jamesaimonetti**
Guardrail to deny any attempt to delete the master account. Both kz_datamgr (via kzs_db) and kzdb_accounts check the configured master account db (from kapps_util) against the db to be deleted.

### CODE REVIEW

**Comment by jamesaimonetti** [comment]
Backports at:
- 5.2 https://github.com/2600hz/kazoo-core/pull/650
- 5.1 https://github.com/2600hz/kazoo-core/pull/651
- 5.0 https://github.com/2600hz/kazoo-core/pull/652

File: kazoo_data/src/kzs_db.erl [hunk 1]
```diff
@@ -99,7 +99,13 @@ do_db_delete_others(Map, DbName) ->
 
 -spec do_db_delete(map(), kz_term:ne_binary()) -> boolean().
 do_db_delete(#{server := {App, Conn}}, DbName) ->
-    App:db_delete(Conn, DbName).
+    case kapps_util:get_master_account_db() of
+        {'ok', DbName} ->
+            lager:error("deleting master account id forbidden"),
+            'false';
+        {'ok', _MasterDb} ->
+            App:db_delete(Conn, DbName)
```
Comment (icehess) [inline kazoo_data/src/kzs_db.erl:107]: Probably at least ignore no master account, no?
```suggestion
        {'ok', _MasterDb} ->
            App:db_delete(Conn, DbName); 
        {'error', 'no_accounts'} ->
            App:db_delete(Conn, DbName); 
        %% {'error', _} ->
        %%    crash?
```

Comment (jamesaimonetti) [reply kazoo_data/src/kzs_db.erl:107]: I honestly can't think of a situation where KAZOO would want to delete a DB before a master account is created.

Comment (jamesaimonetti) [reply kazoo_data/src/kzs_db.erl:107]: I'm good with crashing unless you have a compelling reason to handle the error tuple?

**Review by icehess — APPROVED** [review]

**Pull request merged**